src/com/reason/bs/BsBinaries.java

```diff
@@ -14,44 +14,29 @@
-
-        // First, try 7.2 locations
-        VirtualFile bsbPath = virtualFileManager.findFileByUrl(workingDir + LOCAL_BS_PLATFORM + "/" + platform + "/bsb.exe");
-        if (bsbPath == null) {
-            // Try old locations
-            bsbPath = virtualFileManager.findFileByUrl(workingDir + LOCAL_BS_PLATFORM + "/lib/bsb.exe");
-            if (bsbPath == null) {
-                bsbPath = virtualFileManager.findFileByUrl(workingDir + LOCAL_NODE_MODULES_BIN + "/bsb" + (SystemInfo.isWindows ? ".cmd" : ""));
-            }
-        }
-
-        return bsbPath == null ? null : bsbPath.getPath();
-    }
-
-    // Duplicate BsbPath for more safety
-    @Nullable
-    public static String getBscPath(@NotNull Project project, @NotNull VirtualFile sourceFile) {
-        VirtualFileManager virtualFileManager = VirtualFileManager.getInstance();
-        String platform = getOsBsPrefix();
-        String workingDir = FILE_PROTOCOL_PREFIX + ReasonSettings.getInstance(project).getWorkingDir(sourceFile);
-
-        // First, try 7.2 locations
-        VirtualFile bscPath = virtualFileManager.findFileByUrl(workingDir + LOCAL_BS_PLATFORM + "/" + platform + "/bsc.exe");
-        if (bscPath == null) {
-            // Try old locations
-            bscPath = virtualFileManager.findFileByUrl(workingDir + LOCAL_BS_PLATFORM + "/lib/bsc.exe");
-            if (bscPath == null) {
-                bscPath = virtualFileManager.findFileByUrl(workingDir + LOCAL_NODE_MODULES_BIN + "/bsc" + (SystemInfo.isWindows ? ".cmd" : ""));
-            }
-        }
-
-        return bscPath == null ? null : bscPath.getPath();
-    }
+	@Nullable
+	public static String getBsbPath(@NotNull Project project, @NotNull VirtualFile sourceFile) {
+		VirtualFileManager virtualFileManager = VirtualFileManager.getInstance();
+		String workingDir = FILE_PROTOCOL_PREFIX + ReasonSettings.getInstance(project).getWorkingDir(sourceFile);
+
+		VirtualFile bsbPath = getPathToBinary(virtualFileManager, "bsb", workingDir + LOCAL_BS_PLATFORM);
+		if (bsbPath == null) {
+			bsbPath = getPathToBinary(virtualFileManager, "bsb", workingDir + LOCAL_NODE_MODULES_BIN);
+		}
+		return bsbPath == null ? null : bsbPath.getPath();
+	}
+
+	@Nullable
+	public static String getBscPath(@NotNull Project project, @NotNull VirtualFile sourceFile) {
+		VirtualFileManager virtualFileManager = VirtualFileManager.getInstance();
+		String workingDir = FILE_PROTOCOL_PREFIX + ReasonSettings.getInstance(project).getWorkingDir(sourceFile);
+
+		VirtualFile bscPath = getPathToBinary(virtualFileManager, "bsc", workingDir + LOCAL_BS_PLATFORM);
+		if (bscPath == null) {
+			bscPath = getPathToBinary(virtualFileManager, "bsc", workingDir + LOCAL_NODE_MODULES_BIN);
+		}
+		return bscPath == null ? null : bscPath.getPath();
+	}
 
     @Nullable
     public static String getRefmtPath(@NotNull Project project, @NotNull VirtualFile sourceFile) {
```

Comment:
Different treatment for refmtPath, ist it wanted ? can it use same resolution mechanism getPathToBinary ?

Reply:
yea, refmt has complete different location and name.. so it is easier to leave it inlined as it was